client/bftclient/src/bft_client.cpp

```diff
@@ -70,9 +72,11 @@ Client::Client(SharedCommPtr comm, const ClientConfig& config, std::shared_ptr<c
   communication_->start();
   if (config_.replicas_master_key_folder_path.has_value()) {
     bft::communication::StateControl::instance().setGetPeerPubKeyMethod([&](uint32_t rid) {
+      // The client reads a public key file which the cre client which runs on the same machine maintains
+      // the cre client updates the file after polling updates about replica key exchanges
       concord::secretsmanager::SecretsManagerPlain psm_;
       std::string key_path = config_.replicas_master_key_folder_path.value() + "/" + std::to_string(rid) + "/pub_key";
-      return psm_.decryptFile(key_path).value_or("");
+      return std::array<std::string, replicaIdentityHistoryCount>{psm_.decryptFile(key_path).value_or(""), ""};
```

Comment:
If we keep only the latest key (which makes sense of course) why do we need to return an array of size 2 where the second element is always an empty one?